test/test.pluggableauthclient.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {PluggableAuthClient} from '../src/auth/pluggableauthclient';
+import {BaseExternalAccountClient} from '../src';
+import {getAudience, getTokenUrl} from './externalclienthelper';
+import {beforeEach} from 'mocha';
+
+describe('PluggableAuthClient', () => {
+  const audience = getAudience();
+  const pluggableAuthCredentialSource = {
+    command: './command',
+    output_file: 'output.txt',
+    timeout_millis: 10000,
+  };
+  const pluggableAuthOptions = {
+    type: 'external_account',
+    audience,
+    subject_token_type: 'urn:ietf:params:oauth:token-type:jwt',
+    token_url: getTokenUrl(),
+    credential_source: pluggableAuthCredentialSource,
+  };
+
+  it('should be a subclass of ExternalAccountClient', () => {
+    assert(PluggableAuthClient.prototype instanceof BaseExternalAccountClient);
+  });
+
+  describe('Constructor', () => {
+    it('should throw when credential_source is missing command', () => {
+      const expectedError = new Error(
+        'No valid Pluggable Auth "credential_source" provided.'
+      );
+      const invalidCredentialSource = Object.assign(
+        {},
+        pluggableAuthCredentialSource
+      );
+      delete (invalidCredentialSource as any)['command'];
```

Comment:
nit: this is reasonable in a test, but you should add an eslint ignore line to quiet the linter.